### ISSUE TITLE
https certificate net::ERR_CERT_COMMON_NAME_INVALID on gcping.com

### ISSUE THREAD

**Issue by twodarek** [ImJasonH/gcping#23]
TL;DR: https error on https://gcping.com.  Cert common name says it's for "*.storage.googleapis.com" instead of "gcping.com" or "www.gcping.com"

Hi!

First off, thanks for the useful site!  It's great to see things like this popping up.  🙂 

I noticed that the certificate that the ssl certificate for gcping.com doesn't match the domain name (specifically Chrome saying "net::ERR_CERT_COMMON_NAME_INVALID).

**Steps to repro:**
1. Navigate to https://www.gcping.com/

**Expected outcome:**
2. SSL certificate common name matches and the site appears like magic over ssl

**Actual outcome:**
2. Notice the error page Chrome serves up
3. Inspect the certificate to notice "*.storage.googleapis.com" as the common name instead of "gcping.com"

**Suggested fix:**
Ok, I'm going to be making some guesses based on what I see, so please feel free to correct me or ignore me.

**A**) It looks like you're serving directly from a gStorage bucket (via this https://cloud.google.com/storage/docs/hosting-static-website ?), so I suggest running it through a load balancer following this: https://cloud.google.com/load-balancing/docs/https/adding-a-backend-bucket-to-content-based-load-balancing (and adding ssl on your way through)

**B**) If, instead, you already have a load balancer on the front end (something like this? https://cloud.google.com/load-balancing/docs/https/adding-a-backend-bucket-to-content-based-load-balancing), I'd suggest adding a custom ssl certificate by following the ssl parts of this guide: https://cloud.google.com/load-balancing/docs/https/content-based-example#configuring_the_load_balancing_service

**Remarks**
Again, this is a lot of guessing based on what I'm seeing and I could be completely wrong in my "Suggested fix" section.  Let me know if I can help in any way!

**Comment by imjasonh** [comment]
Thanks for the report!

Yeah, HTTPS is just borked in general (see #22 for more info). The site is currently served through GCS but it doesn't have any HTTPS LB in front -- if it did, the HTTP(non-S) requests it makes to ping backends would be flagged as insecure content, and rightly so.

**Comment by imjasonh** [comment]
Fixed by https://github.com/ImJasonH/gcping/issues/29
